### PR TITLE
choke: discard peerRequests optimization

### PR DESCRIPTION
**Pull request by astro**
this just caught my eye.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -187,7 +189,7 @@ Wire.prototype.handshake = function (infoHash, peerId, extensions) {
 Wire.prototype.choke = function () {
   if (this.amChoking) return
   this.amChoking = true
-  while (this.peerRequests.length) this.peerRequests.pop()
+  this.peerRequests.slice(0, this.peerRequests.length)
```
Comment (feross) [inline index.js]: `slice` doesn't change the array. It just returns a shallow copy while leaving the original array in tact.

I think you wanted `splice`. Or, how about we just do: `this.peerRequests = []`

Comment (astro) [inline index.js]: oops, bad typo

**Comment by astro** [comment]
I assume the goal here was to keep valid any references to the array.

**Comment by feross** [comment]
Oh, correct.

**Comment by feross** [comment]
Merged and published as 1.5.3.

**Pull request merged**